peerprep-fe/src/app/collaboration/components/CollaborationEditor.tsx

```diff
@@ -0,0 +1,261 @@
+'use client';
+
+import { useEffect, useRef, useState } from 'react';
+import * as Y from 'yjs';
+import { WebsocketProvider } from 'y-websocket';
+import { MonacoBinding } from 'y-monaco';
+import { editor as MonacoEditor } from 'monaco-editor';
+import { OnMount } from '@monaco-editor/react';
+import { useAuthStore } from '@/state/useAuthStore';
+import Avatar, { genConfig } from 'react-nice-avatar';
+import CodeEditor from './CodeEditor';
+import { Button } from '@/components/ui/button';
+import {
+  Select,
+  SelectTrigger,
+  SelectValue,
+  SelectItem,
+  SelectContent,
+} from '@/components/ui/select';
+import { AwarenessState, ConnectedClient } from '@/types/types';
+import { useToast } from '@/hooks/use-toast';
+import { useCollaborationStore } from '@/state/useCollaborationStore';
+import {
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+  AlertDialogTrigger,
+} from '@/components/ui/alert-dialog';
+import { useRouter } from 'next/navigation';
+
+interface CollaborationEditorProps {
+  language: string;
+  matchId: string | null;
+  onLanguageChange: (language: string) => void;
+  supportedLanguages: string[];
+}
+
+const CollaborationEditor = ({
+  language,
+  matchId,
+  onLanguageChange,
+  supportedLanguages,
+}: CollaborationEditorProps) => {
+  const { user } = useAuthStore();
+  const [connectedClients, setConnectedClients] = useState<
+    Map<number, ConnectedClient>
+  >(new Map());
+  const providerRef = useRef<WebsocketProvider | null>(null);
+  const bindingRef = useRef<MonacoBinding | null>(null);
+  const editorRef = useRef<MonacoEditor.IStandaloneCodeEditor | null>(null);
+  const prevClientsRef = useRef<Map<number, ConnectedClient>>(new Map());
+  const sockServerURI =
+    process.env.NEXT_PUBLIC_SOCK_SERVER_URL || 'ws://localhost:4444';
+  const { toast } = useToast();
+  const { clearLastMatchId } = useCollaborationStore();
+  const router = useRouter();
+
+  const stringToColor = (str: string) => {
```

Comment:
minor nit: you might want to shift this code into utils to allow for reusability